astropy/io/ascii/ui.py

```diff
@@ -324,7 +327,6 @@ def read(table, guess=None, **kwargs):
             guess = False
 
     if not guess:
-        reader = get_reader(**new_kwargs)
         # Try the fast reader version of `format` first if applicable.  Note that
         # if user specified a fast format (e.g. format='fast_basic') this test
         # will fail and the else-clause below will be used.
```

Comment:
Contrary to the comment, the `get_reader()` call here always first created a `basic.Basic` reader instance, which would still stick around if the subsequent attempt to get a fast version failed (e.g. if no fast version of the requested format exists).